app/uk/gov/hmrc/transitmovementsrouter/controllers/MessagesController.scala

```diff
@@ -135,19 +138,36 @@ class MessagesController @Inject() (
     }
   }
 
+  private def extractAuditMessageType(messageType: MessageType): EitherT[Future, MessageTypeExtractionError, AuditType] = {
+    val auditType: EitherT[Future, MessageTypeExtractionError, AuditType] =
+      EitherT.fromOption[Future](messageType.auditType, MessageTypeExtractionError.InvalidMessageType(s"$messageType is not a ResponseMessageType"))
+    auditType
```

Comment:
Get rid of the local variable, this can be a one liner -- though really, as this is in the presentation layer, instead of returning a `MessageTypeExtractionError`, just return a `PresentationError` directly.